src/LiveSplit.Core/Model/RunMetadata.cs

```diff
@@ -117,6 +117,49 @@ public IDictionary<Variable, VariableValue> VariableValues
             }
         }
 
+        public class CustomVariable
+        {
+            public string Value = null;
+            public bool IsPermanent = false;
```

Comment:
Does this really need to be mutable?  
Do not expose fields publicly.

```suggestion
            public bool IsPermanent { get; set; }
```